kernel/Kconfig

```
@@ -1020,6 +1020,12 @@ config THREAD_LOCAL_STORAGE
 	help
 	  This option enables thread local storage (TLS) support in kernel.
 
+config KERNEL_WHOLE_ARCHIVE
+	bool "Link Kernel Library as whole archive"
```

it's good to get the control of kernel lib / whole-archive handling decoupled from llext, but what is the reason / use-case for the `KERNEL_WHOLE_ARCHIVE` to have a prompt, meaning it suddenly becomes user configurable.

Initial feeling is that this symbol should be promptless and thereby only be selected by subsystems like llext.
If an app has special reasons why kernel should be included in whole-archive, then an app `Kconfig` with a `select KERNEL_WHOLE_ARCHIVE` is also safer than a `CONFIG_KERNEL_WHOLE_ARCHIVE=y` in a prj.conf.

Feel free to provide an explanation if you believe the prompt should be kept.

```suggestion
	bool
```

---

Fair point, that is how I'm using in my app as well.

I don't have a strong reason why it should be prompt-able.
My initial thought was that a simple app may select it's configuration directly in `prj.conf` without requiring an app `Kconfig`.
Apps built with this explicit requirement are likely to have this as a fundamental constant enablement.

I'm fine to remove the prompt.

---

I believe the best in this case is to then have it promptless in this case.

---

Apologies for the delay, the prompt has now been removed.